src/ngraph/op/deformable_psroi_pooling.cpp

```diff
@@ -0,0 +1,157 @@
+                                                       const Output<Node>& coords,
+                                                       const int64_t output_dim,
+                                                       const int64_t group_size,
+                                                       float spatial_scale,
+                                                       const std::string mode,
+                                                       int64_t spatial_bins_x,
+                                                       int64_t spatial_bins_y,
+                                                       bool no_trans,
+                                                       float trans_std,
+                                                       int64_t part_size)
+    : Op({input, coords})
+    , m_output_dim(output_dim)
+    , m_group_size(group_size)
+    , m_spatial_scale(spatial_scale)
+    , m_mode(mode)
+    , m_spatial_bins_x(spatial_bins_x)
+    , m_spatial_bins_y(spatial_bins_y)
+    , m_no_trans(no_trans)
+    , m_trans_std(trans_std)
+    , m_part_size(part_size)
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::v1::DeformablePSROIPooling::validate_and_infer_types()
+{
+    const auto& input_et = get_input_element_type(0);
+
+    const auto& input_pshape = get_input_partial_shape(0);
+    const auto& box_coords_pshape = get_input_partial_shape(1);
+
+    NODE_VALIDATION_CHECK(this,
+                          input_pshape.rank().is_dynamic() ||
+                              static_cast<size_t>(input_pshape.rank()) == 4,
+                          "Feature map input rank must equal to 4 (input rank: ",
+                          static_cast<size_t>(input_pshape.rank()),
+                          ")");
+    NODE_VALIDATION_CHECK(this,
+                          box_coords_pshape.rank().is_dynamic() ||
+                              static_cast<size_t>(box_coords_pshape.rank()) == 2,
+                          "Box coordinates input rank must equal to 2 (input rank: ",
+                          static_cast<size_t>(box_coords_pshape.rank()),
+                          ")");
+
+    if (get_input_size() == 3) // offsets input is provided
+    {
+        const auto& offsets_pshape = get_input_partial_shape(2);
+        NODE_VALIDATION_CHECK(this,
+                              offsets_pshape.rank().is_dynamic() ||
+                                  static_cast<size_t>(offsets_pshape.rank()) == 4,
+                              "Offsets input rank must equal to 4 (input rank: ",
+                              static_cast<size_t>(offsets_pshape.rank()),
+                              ")");
+    }
+    int64_t output_rank = 4;
+    std::vector<Dimension> output_dim_vec(output_rank, Dimension::dynamic());
+    if (box_coords_pshape[0].is_static())
+    {
+        output_dim_vec[0] = box_coords_pshape.to_shape()[0];
+    }
+    output_dim_vec[1] = m_output_dim;
+    for (int i = 2; i < 4; ++i)
```

Comment:
Shouldn't `output_rank` be used in this loop?

Reply:
Right, thanks